pkg/scheduler/queue/tenant_querier_assignment.go

```diff
@@ -249,20 +251,19 @@ func (tqa *tenantQuerierAssignments) setup(dequeueArgs *DequeueArgs) {
 // Note that because we use the shared  tenantIDOrder and tenantOrderIndex to manage the queue, we functionally
 // ignore each Node's individual queueOrder and queuePosition.
 func (tqa *tenantQuerierAssignments) dequeueSelectNode(node *Node) *Node {
+	if node.isLeaf() || len(node.queueMap) == 0 {
```

Comment:
`len(node.queueMap) == 0` should never occur for a non-leaf node, correct? any empty paths would get deleted up until the nodes have children again

Reply:
Yeah, this code is _technically_ unreachable under normal conditions, since leaf nodes also get checked for earlier in dequeue. I was torn on whether to leave this check in or take it out, and I'm still not strongly opinionated either way; I just left it because it seemed like the _especially_ dummy-proof thing to do.

Reply:
I am fine with that; I lean more towards "remove the safeguards and investigate if it breaks" because dummy-proofing is sort of a never-ending rabbit hole, but since we are still in rollout we can leave for now.